pycqed/instrument_drivers/physical_instruments/QuTech_VSM_Module.py

```diff
@@ -0,0 +1,267 @@
+        idn.update(self.get(idn_param))
+        t = time.time() - (begin_time or self._t0)
+
+        logging.info('Connected to: {vendor} {model} '
+                     '(serial:{serial}, firmware:{firmware}) '
+                     'in {t:.2f}s'.format(t=t, **idn))
+
+    ###
+    # Generic SCPI commands
+    ###
+
+    def get_identity(self):
+        return self.ask('*IDN?')
+
+    def get_operation_complete(self):
+        return self.ask('*OPC?')
+
+    def get_error(self):
+        """ Returns:    "No error" or <error message>
+        """
+        return self.ask('system:error?')
+
+    def get_system_error_count(self):
+        return self.ask_int('system:error:count?')
+
+    # Parameters
+
+    def add_power_supply_parameters(self):
+        for power_supply in ['front', 'rear', 'phantom']:
+            name = 'powersupply_{p}'.format(p=power_supply)
+            getter = 'POWERSUPPLY:{p}?'.format(p=power_supply.upper())
+            setter = 'POWERSUPPLY:{p}'.format(p=power_supply.upper()) + ' {}'
+            # Disable setting the front power supply
+            if power_supply == 'front':
+                setter = None
+            self.add_parameter(name, get_cmd=getter, set_cmd=setter,
+                               vals=vals.OnOff())
+
+    def add_temperature_parameters(self):
+        # Command class temperature
+        self.add_parameter('setpoint',
+                           get_cmd='TEMPERATURE:SETPOINT?',
+                           set_cmd='TEMPERATURE:SETPOINT {}',
+                           vals=vals.Numbers(min_value=10.0, max_value=55.0))
+        self.add_parameter('temperature_avg',
+                           get_cmd='TEMPERATURE?',
+                           vals=vals.Numbers())
+
+        for mod in self.modules:
+            mod_name = 'mod{m}'.format(m=mod)
+            mod_scpi = 'MODULE{m}'.format(m=mod)
+            self.add_parameter(mod_name + '_temperature_avg',
+                               get_cmd='TEMPERATURE:'+mod_scpi+'?',
+                               vals=vals.Numbers())
+            self.add_parameter(mod_name + '_temperature_digital',
+                               get_cmd='TEMPERATURE:' + mod_scpi + ':DIGITAL?',
+                               vals=vals.Numbers())
+        for channel in self.channels:
+            for mod in self.modules:
+                mod_ch_name = 'mod{m}_ch{c}'.format(m=mod, c=channel)
+                mod_ch_scpi = 'MODULE{m}:CHANNEL{c}'.format(m=mod, c=channel)
+                self.add_parameter(mod_ch_name + '_temperature',
+                                   get_cmd='TEMPERATURE:'+mod_ch_scpi+'?',
+                                   vals=vals.Numbers())
+
+    def add_marker_parameters(self):
+        # Command class marker
+        self.add_parameter('marker_source',
+                           set_cmd='MARKER:SOURCE {}',
+                           vals=vals.Enum('int', 'ext'))
+        self.add_parameter('marker_state',
```

Comment:
The difference between marker state and marker_source is unclear to me. Could you add a docstring?